presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushProjectionThroughExchange.java

```diff
@@ -150,7 +147,7 @@
         return Optional.of(restrictOutputs(context.getIdAllocator(), result, ImmutableSet.copyOf(project.getOutputSymbols())).orElse(result));
     }
 
-    private boolean isSymbolToSymbolProjection(ProjectNode project)
+    private static boolean isSymbolToSymbolProjection(ProjectNode project)
```

Comment:
`ProjectNode::isIdentity` is stricter than this (it also checks expression name = symbol name). I agree that changes not related to pattern matching should be done out of this PR - just like the `isDistinct` change I'm going to split out of this. Feel free to open a PR changing that rule.